internal/pkg/policy/policy_output.go

```diff
@@ -246,6 +271,71 @@ func (p *Output) prepareElasticsearch(
 	return nil
 }
 
+func (p *Output) createAndGetBulker(ctx context.Context, mainBulker bulk.Bulk, outputMap map[string]map[string]interface{}) (bulk.Bulk, error) {
+	bulker := bulkerMap[p.Name]
+	if bulker != nil {
+		// TODO replace if hosts or service token changed?
+		return bulker, nil
+	}
+	bulkCtx, bulkCancel := context.WithCancel(context.Background())
+	defer bulkCancel()
+	es, err := p.createRemoteEsClient(bulkCtx, outputMap)
+	if err != nil {
+		return nil, err
+	}
+	// starting a new bulker to create/update API keys for remote ES output
+	newBulker := bulk.NewBulker(es, nil, mainBulker.Opts())
+	bulkerMap[p.Name] = newBulker
+
+	errCh := make(chan error)
+	go func() {
+		runFunc := func() (err error) {
+			return newBulker.Run(bulkCtx)
+		}
+
+		errCh <- runFunc()
+	}()
+	go func() (err error) {
+		select {
+		case err = <-errCh:
+		case <-bulkCtx.Done():
+			err = bulkCtx.Err()
+		}
+		return
+	}()
+
+	return newBulker, nil
+}
+
+func (p *Output) createRemoteEsClient(ctx context.Context, outputMap map[string]map[string]interface{}) (*elasticsearch.Client, error) {
+	hostsObj := outputMap[p.Name]["hosts"]
+	hosts, ok := hostsObj.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("failed to get hosts from output: %v", hostsObj)
+	}
+	hostsStrings := make([]string, len(hosts))
+	for i, host := range hosts {
+		hostsStrings[i], ok = host.(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to get hosts from output: %v", host)
+		}
+	}
+
+	cfg := config.Config{
+		Output: config.Output{
+			Elasticsearch: config.Elasticsearch{
+				Hosts:        hostsStrings,
+				ServiceToken: p.ServiceToken,
+			},
+		},
+	}
+	es, err := es.NewClient(ctx, &cfg, false)
```

Comment:
We should get APM traces

Reply:
Do you mean like here? I'm trying to add it, not sure how to pass the bi (build info) from Fleet

https://github.com/elastic/fleet-server/blob/a2f232939f961887a511a5434d84d7c0778e59c5/internal/pkg/server/fleet.go#L458-L459

Reply:
I think that bi is used to attach the fleet-server user agent header: https://github.com/elastic/fleet-server/blob/main/internal/pkg/es/client.go#L53